app/services/slack_notifier_service.rb

```diff
@@ -0,0 +1,34 @@
+class SlackNotifierService
+  CHANNEL = '#tariffs-etl'.freeze
+  USERNAME = 'Trade Tariff BE'.freeze
+
+  class << self
+    def call(message, msg_notifier:)
+      @msg_notifier = msg_notifier
+
+      if slack_web_hook_url.blank?
+        logger.warn('The ENV "SLACK_WEB_HOOK_URL" is not declared. Slack notifications are disabled.')
+
+        return
+      end
+
+      notifier.ping(message)
+    end
+
+  private
+
+    def notifier
+      @msg_notifier || slack_notifier
```

Comment:
I am injecting the  @msg_notifier to simplify tests, even though I could probably use allowAnyInstanceOf (which is deprecated anyway)

Reply:
I wonder if it would be better to configure slack notifier as part of an initialiser for this gem?

Reply:
That also feels like a better place to access environment variables